src/traces/splom/base_plot.js

```diff
@@ -168,46 +167,40 @@ function makeGridData(gd) {
     return gridBatches;
 }
 
-function clean(newFullData, newFullLayout, oldFullData, oldFullLayout, oldCalcdata) {
-    var oldModules = oldFullLayout._modules || [];
-    var newModules = newFullLayout._modules || [];
+function clean(newFullData, newFullLayout, oldFullData, oldFullLayout) {
+    oldLoop:
+    for(var i = 0; i < oldFullData.length; i++) {
+        var oldTrace = oldFullData[i];
 
-    var hadSplom, hasSplom;
-    var i;
+        if(oldTrace.type === 'splom') {
+            for(var j = 0; j < newFullData.length; j++) {
+                var newTrace = newFullData[j];
 
-    for(i = 0; i < oldModules.length; i++) {
-        if(oldModules[i].name === 'splom') {
-            hadSplom = true;
-            break;
-        }
-    }
-    for(i = 0; i < newModules.length; i++) {
-        if(newModules[i].name === 'splom') {
-            hasSplom = true;
-            break;
-        }
-    }
+                if(oldTrace.uid === newTrace.uid && newTrace.type === 'splom') {
+                    continue oldLoop;
+                }
+            }
```

Comment:
unlikely to matter much, as having a large number of traces on a splom plot would be weird... but you could reduce this to two separate loops, one to collect the `uid` of each `newFullData` `splom` trace into an object `newUids = {uid1: 1, uid2: 1}` and the second would just do `if(newUids[oldTrace.uid]) continue`

Reply:
Thanks for the tip. Done -> https://github.com/plotly/plotly.js/pull/3057/commits/478c6693ecb639f8b718b0ab5a608cefd6920c55